platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/file/FileHandler.java

```diff
@@ -0,0 +1,114 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.logging.file;
+
+import com.swirlds.config.api.Configuration;
+import com.swirlds.logging.api.Level;
+import com.swirlds.logging.api.extensions.emergency.EmergencyLogger;
+import com.swirlds.logging.api.extensions.emergency.EmergencyLoggerProvider;
+import com.swirlds.logging.api.extensions.event.LogEvent;
+import com.swirlds.logging.api.extensions.handler.AbstractSyncedHandler;
+import com.swirlds.logging.api.internal.format.LineBasedFormat;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.BufferedWriter;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.file.Path;
+import java.util.Objects;
+
+public class FileHandler extends AbstractSyncedHandler {
+
+    private static final String FILE_NAME_PROPERTY = "%s.file";
+    private static final String BUFFER_SIZE_PROPERTY = "%s.bufferSize";
+    private static final String APPEND_PROPERTY = "%s.append";
+    private static final String DEFAULT_FILE_NAME = "swirlds-log.log";
+
+    /**
+     * The emergency logger.
+     */
+    private static final EmergencyLogger EMERGENCY_LOGGER = EmergencyLoggerProvider.getEmergencyLogger();
+
+    private final FileOutputStream fileOutputStream;
+    private final OutputStreamWriter outputStreamWriter;
+    private final BufferedWriter bufferedWriter;
+
+    /**
+     * Creates a new file handler.
+     *
+     * @param configKey     the configuration key
+     * @param configuration the configuration
+     */
+    public FileHandler(@NonNull String configKey, @NonNull Configuration configuration) {
+        super(configKey, configuration);
+
+        final String propertyPrefix = PROPERTY_HANDLER.formatted(configKey);
+        final Path filePath = Objects.requireNonNullElse(
+                configuration.getValue(FILE_NAME_PROPERTY.formatted(propertyPrefix), Path.class, null),
+                Path.of(DEFAULT_FILE_NAME));
+        final int bufferSize = Objects.requireNonNullElse(
+                configuration.getValue(BUFFER_SIZE_PROPERTY.formatted(propertyPrefix), Integer.class, null), 8 * 1024);
+        final boolean append = Objects.requireNonNullElse(
+                configuration.getValue(APPEND_PROPERTY.formatted(propertyPrefix), Boolean.class, null), true);
+
+        FileOutputStream fileOutputStream = null;
+        OutputStreamWriter outputStreamWriter = null;
+        BufferedWriter bufferedWriter = null;
+        try {
+            fileOutputStream = new FileOutputStream(filePath.toFile(), append);
+            outputStreamWriter = new OutputStreamWriter(fileOutputStream);
+            bufferedWriter = new BufferedWriter(outputStreamWriter, bufferSize);
+        } catch (final Exception exception) {
+            EMERGENCY_LOGGER.log(Level.ERROR, "Failed to create FileHandler", exception);
+        }
```

Comment:
Hmmm...  From what I can tell we only actually need/use the `BufferedWriter` object (the other two are underlying details).
Would it be reasonable, here, to use something _similar to_ the following and leave those underlying details to the JVM classes?
```suggestion
        try {
            if (!Files.exists(path) || Files.isWritable(path)) {
                if (append) {
                    bufferedWriter = Files.newBufferedWriter(filePath, StandardOpenOption.CREATE, StandardOpenOption.APPEND, StandardOpenOption.WRITE, StandardOpenOption.DSYNC);
                } else {
                    bufferedWriter = Files.newBufferedWriter(filePath, StandardOpenOption.CREATE, StandardOpenOption.WRITE, StandardOpenOption.DSYNC);
                }
            } else {
                // Log appropriately that the file exists and is not writable
            }
        } catch (final Exception exception) {
            EMERGENCY_LOGGER.log(Level.ERROR, "Failed to create FileHandler", exception);
        }
```

Reply:
I wanted to provide an easy way to specify the buffer size. From what I can see, your approach does not allow for this.

Reply:
True, we wouldn't be able to specify the buffer size, then again are we certain we actually need to?  In the extreme majority of situations I've seen, changing the buffer size from the default isn't necessary or helpful.
With DSYNC (or flush-after-write to manually do the same thing), which the documentation suggests is used here, a larger or smaller buffer is almost irrelevant.

Reply:
I would say that this is a good point for JMH or something to check what is the best solution. From an api point of view I'm with @jsync-swirlds . But based on the comment from @timo0 other logging frameworks do not use it. So we should find the reason behind that.

Next to that I would prefer to provide config variables only if we really need them. Based on that I would remove the buffer size as a config variable + set it as a fixed value in the code.